src/shared/utilities/collectionUtils.ts

```diff
@@ -230,3 +230,21 @@ export class IteratorTransformer<TIteratorOutput, TTransformerOutput> {
         }
     }
 }
+
+/**
+ * Push if condition is true, useful for adding CLI arguments, and avoiding this kind of sitatuion:
+ * if (x && y) {
+ *     arr.push(item2)
+ *     if(z) {
+ *         arr.push(item)
+ *     }
+ * }
+ * @param args The array to push to
+ * @param addIfConditional conditional that determines if we will push to the array
+ * @param argsToAdd The additional items to append to the array
+ */
+export function pushIf(args: string[], addIfConditional: boolean, ...argsToAdd: string[]) {
```

Comment:
nit: This file seems to contain general purpose collection functionality. Can this function be generalized to remove the string type and the "args" naming?

Looking at this from a more generalized lens, would the "addIf" param make more sense as a lambda? (I don't think I'd want to change the existing calls into lambdas though)

Reply:
Made it generic (we could have done that later as needed though), a lambda would make sense for Kotlin, but doesn't really yield a benefit for Typescript IMO (since we don't have nice higher order function syntax) (we can use `?.let` in Kotlin so we wouldn't even have this function but anyway)